File: frontend/app/validators/columnValidator.ts
import * as yup from "yup";

export const columnValidator = yup.object({
  name: yup
    .string()
    .required("Название колонки обязательно")
    .min(3, "Название должно быть не менее 3 символов")
    .max(50, "Название должно быть не более 50 символов"),
  isFinished: yup.boolean().required(),
});

File: backend/backend.Application/Validators/ColumnValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using backend.Application.DTOs;
using FluentValidation;

namespace backend.Application.Validators
{
    public class ColumnValidator : AbstractValidator<ColumnDTO>
    {
        public ColumnValidator()
        {
            RuleFor(x => x.Name)
                .NotEmpty().WithMessage("Name is required")
                .MinimumLength(3).WithMessage("Name must be at least 3 characters long")
                .MaximumLength(50).WithMessage("Name must be at most 50 characters long");
        }
    }
}